pkg/cloud/services/network/natgateways.go

```diff
@@ -287,7 +287,10 @@ func (s *Service) deleteNatGateway(id string) error {
 		}
 
 		if len(out.NatGateways) == 0 {
-			return false, errors.Wrapf(err, "no NAT gateway returned for id %q", id)
+			if err != nil {
```

Comment:
nit: remove this err check, as we are already checking it at [Line285](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3186/files#diff-23c065949155c8a08cd81009d31c08a2d86909abe59c908a374d11c9a641281eL285)
```
if err != nil {
   return false, errors.Wrapf(err, "no NAT gateway returned for id %q", id)
}
```
/lgtm